doc/source/whatsnew/v2.3.0.rst

```diff
@@ -50,10 +50,20 @@ Notable bug fixes
 
 These are bug fixes that might have notable behavior changes.
 
-.. _whatsnew_230.notable_bug_fixes.notable_bug_fix1:
+.. _whatsnew_230.notable_bug_fixes.string_comparisons:
 
-notable_bug_fix1
-^^^^^^^^^^^^^^^^
+Comparisons between different string dtypes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In previous versions, comparing Series of different string dtypes (e.g. ``pd.StringDtype("pyarrow", na_value=pd.NA)`` against ``pd.StringDtype("python", na_value=np.nan)``) would result in inconsistent resulting dtype or incorrectly raise. pandas will now use the hierarchy
+
+    object < (python, NaN) < (pyarrow, NaN) < (python, NA) < (pyarrow, NA)
+
+in determining the result dtype when there are different string dtypes compared. Some examples:
+
+- When ``pd.StringDtype("pyarrow", na_value=pd.NA)`` is compared against any other string dtype, the result will always be ``boolean[pyarrow]``.
```

Comment:
I think this is not correct, it returns just the nullable `boolean` dtype? (i.e. `pd.BooleanDtype()`) Where `boolean[pyarrow]` is an alias for `pd.ArrowDtype(pa.boolean())`

Reply:
And I also see that I coded explicitly this expected dtype in the tests ...

Reply:
Given the ordering

    object < (python, NaN) < (pyarrow, NaN) < (python, NA) < (pyarrow, NA)

when we compare `(pyarrow, NA)` with anything we want the result to be `(pyarrow, NA)`, which is `boolean[pyarrow]`.